core/capability-negotiation.md

```diff
@@ -0,0 +1,457 @@
+
+Once capability negotiation has completed with a client-send `CAP END` command, registration
+continues as normal.
+
+With the above registration process, clients will either recieve a `CAP` message indicating
+that registration is paused while capability negotiation happens, or registration will
+complete immediately, indicating that the server doesn't support capability negotiation.
+
+Example with capability negotiation:
+
+    Client: CAP LS 302
+    Client: NICK dan
+    Client: USER d * 0 :This is a really good name
+    Server: CAP * LS :multi-prefix sasl
+    Client: CAP REQ :multi-prefix
+    Server: CAP * ACK multi-prefix
+    Client: CAP END
+    Server: 001 dan :Welcome to the Internet Relay Network dan
+    ...
+
+Example with capability negotiation, but where the client recognises no advertised caps: 
+
+    Client: CAP LS 302
+    Client: NICK dan
+    Client: USER d * 0 :This is a really good name
+    Server: CAP * LS :draft/example-1 draft/example-2
+    Client: CAP END
+    Server: 001 dan :Welcome to the Internet Relay Network dan
+    ...
+
+Example where the server doesn't support capability negotiation:
+
+    Client: CAP LS 302
+    Client: NICK dan
+    Client: USER d * 0 :This is a really good name
+    Server: 001 dan :Welcome to the Internet Relay Network dan
+    ...
+
+Example where the client doesn't support capability negotiation:
+
+    Client: NICK dan
+    Client: USER d * 0 :This is a really good name
+    Server: 001 dan :Welcome to the Internet Relay Network dan
+    ...
+
+## The CAP command
+
+The client capability negotiation extension is implemented by the addition of one command
+with several subcommands.  The command added is named `CAP`.  `CAP` takes a single
+required subcommand, optionally followed by a single parameter of space-separated capability
+identifiers.  Each capability in the list may be preceded by a capability modifier.
+
+The subcommands for `CAP` are: `LS`, `LIST`, `REQ`, `ACK`, `NAK`, and `END`.
+
+The `LS`, `LIST`, `REQ`, `ACK` and `NAK` subcommands may be followed by a single parameter
+containing a space-separated list of capabilities.  If more than one capability is named,
+the RFC1459 designated sentinel (`:`) for a multi-parameter argument must be present. The
+list of capabilities MUST be parsed from left to right and capabilities SHOULD only be sent
+once per command. If a capability is sent multiple times, the last one received takes priority.
+
+If a client sends a subcommand which is not in the list above or otherwise issues an
+invalid command, then numeric 410 (ERR_INVALIDCAPCMD) should be sent.  The first parameter
+after the client identifier (usually nickname) should be the commandname; the second parameter
+should be a human-readable description of the error.
+
+Replies from the server must contain the client identifier name or asterisk if one is not yet
+available.
+
+Example:
+
+    Client: CAP FOO
+    Server: :example.org 410 * FOO :Invalid CAP command
+
```

Comment:
Include example of a server response, with nick/asterisk.

Reply:
Added an example with a nickname